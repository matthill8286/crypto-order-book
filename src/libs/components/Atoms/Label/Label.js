import * as React from 'react';
export var Label = function Label(_ref) {
  var children = _ref.children,
      htmlFor = _ref.htmlFor;
  return React.createElement("label", {
    htmlFor: htmlFor
  }, children);
};
//# sourceMappingURL=Label.js.map