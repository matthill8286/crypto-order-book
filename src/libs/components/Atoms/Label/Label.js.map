{"version":3,"sources":["../../../../src/components/Atoms/Label/Label.tsx"],"names":["React","Label","children","htmlFor"],"mappings":"AAAA,OAAO,KAAKA,KAAZ,MAAuB,OAAvB;AAMA,OAAO,IAAMC,KAA2B,GAAG,SAA9BA,KAA8B,OAA2B;AAAA,MAAxBC,QAAwB,QAAxBA,QAAwB;AAAA,MAAdC,OAAc,QAAdA,OAAc;AACpE,SAAO;AAAO,IAAA,OAAO,EAAEA;AAAhB,KAA0BD,QAA1B,CAAP;AACD,CAFM","sourcesContent":["import * as React from 'react'\n\nexport interface LabelProps {\n  htmlFor: string\n}\n\nexport const Label: React.FC<LabelProps> = ({ children, htmlFor }) => {\n  return <label htmlFor={htmlFor}>{children}</label>\n}\n"],"file":"Label.js"}