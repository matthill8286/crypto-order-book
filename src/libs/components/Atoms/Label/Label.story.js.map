{"version":3,"sources":["../../../../src/components/Atoms/Label/Label.story.tsx"],"names":["storiesOf","React","Label","Readme","stories","module","options","info","add"],"mappings":"AAAA,SAASA,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,KAAT;AACA,OAAOC,MAAP;AAEA,IAAMC,OAAO,GAAGJ,SAAS,CAAC,2BAAD,EAA8BK,MAA9B,CAAzB;AACA,IAAMC,OAAO,GAAG;AAAEC,EAAAA,IAAI,EAAEJ;AAAR,CAAhB;AAEAC,OAAO,CAACI,GAAR,CAAY,SAAZ,EAAuB;AAAA,SAAM,oBAAC,KAAD;AAAO,IAAA,OAAO,EAAC;AAAf,eAAN;AAAA,CAAvB,EAAuEF,OAAvE","sourcesContent":["import { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { Label } from './Label'\nimport Readme from './Label.readme.md'\n\nconst stories = storiesOf('Design System/Atoms/Label', module)\nconst options = { info: Readme }\n\nstories.add('Default', () => <Label htmlFor=\"field-1\">A label</Label>, options)\n"],"file":"Label.story.js"}