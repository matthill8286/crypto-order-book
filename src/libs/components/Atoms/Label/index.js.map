{"version":3,"sources":["../../../../src/components/Atoms/Label/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './Label'\n"],"file":"index.js"}