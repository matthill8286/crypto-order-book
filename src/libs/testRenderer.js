import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map