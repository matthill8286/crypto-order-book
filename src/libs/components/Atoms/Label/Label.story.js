import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { Label } from "./Label";
import Readme from "./Label.readme.md";
var stories = storiesOf('Design System/Atoms/Label', module);
var options = {
  info: Readme
};
stories.add('Default', function () {
  return React.createElement(Label, {
    htmlFor: "field-1"
  }, "A label");
}, options);
//# sourceMappingURL=Label.story.js.map