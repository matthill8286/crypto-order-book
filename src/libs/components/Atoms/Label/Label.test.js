import React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { Label } from "./Label";
describe('<Label> Atom', function () {
  test('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(Label, {
      htmlFor: "filed-1"
    }, "A label"));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=Label.test.js.map