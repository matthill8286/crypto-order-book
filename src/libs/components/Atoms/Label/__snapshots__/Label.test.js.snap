// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label> Atom renders correctly 1`] = `
<label
  htmlFor="filed-1"
>
  A label
</label>
`;
